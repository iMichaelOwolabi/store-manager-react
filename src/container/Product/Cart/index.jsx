/* eslint-disable react/jsx-one-expression-per-line */
import React, { Fragment } from 'react';
import { Header, Table, Image } from 'semantic-ui-react';
import AuthNav from '../../../components/AuthNav';
import './Cart.scss';

const choiceProduct = JSON.parse(sessionStorage.getItem('cart'));

class Cart extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      products: {
        quantity: sessionStorage.getItem('qty')
      }
    };
  }

  handleChange = (event) => {
    const { products } = this.state;
    const { name, value } = event.target;
    products[name] = value;
    this.setState({
      products
    });
    sessionStorage.setItem('qty', products.quantity);
  }

  render() {
    const { products } = this.state;
    return (
      <Fragment>
        <AuthNav />
        <div className="wrapper">
          <div className="cartHeader" />
          <h2 className="topSpace">Your Shopping Cart</h2>
          <Table celled padded>
            <Table.Header>
              <Table.Row>
                <Table.HeaderCell>S/N</Table.HeaderCell>
                <Table.HeaderCell>Picture</Table.HeaderCell>
                <Table.HeaderCell>Quantity</Table.HeaderCell>
                <Table.HeaderCell>Price</Table.HeaderCell>
                <Table.HeaderCell>Total</Table.HeaderCell>
              </Table.Row>
            </Table.Header>

            <Table.Body>
              <Table.Row>
                <Table.Cell>
                  <Header as="h2">1</Header>
                </Table.Cell>
                <Table.Cell>
                  <Image src={choiceProduct.productimage} rounded size="mini" />
                </Table.Cell>
                <Table.Cell><input type="number" value={products.quantity} name="quantity" className="qty" onChange={this.handleChange} /></Table.Cell>
                <Table.Cell>&#8358; {choiceProduct.price}</Table.Cell>
                <Table.Cell>
                  &#8358;
                  {parseFloat(choiceProduct.price * products.quantity)}
                </Table.Cell>
              </Table.Row>
            </Table.Body>
          </Table>
          <button type="submit" className="salesButton">Make Sales</button>
        </div>
      </Fragment>
    );
  }
}

export default Cart;
