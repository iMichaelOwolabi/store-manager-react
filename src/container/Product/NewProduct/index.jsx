import React, { Fragment } from 'react';
import AuthNav from '../../../components/AuthNav';

const NewProduct = () => (
  <Fragment>
    <AuthNav />
    <div className="container">
      New Product
    </div>
  </Fragment>
);
export default NewProduct;
