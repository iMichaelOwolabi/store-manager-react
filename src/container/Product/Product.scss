@import '../../partial-styles/colors.scss';
.ui.basic.button.newProduct, .ui.basic.buttons .button {
  background: transparent none !important;
  color: $maincolor;
  font-weight: 400;
  border-radius: .28571429rem;
  text-transform: none;
  text-shadow: none !important;
  box-shadow: 0 0 0 1px rgba(34, 36, 38, 0.15) inset;
  border-style: double;
  border-color: $maincolor;
  margin-top: 1rem;
}
