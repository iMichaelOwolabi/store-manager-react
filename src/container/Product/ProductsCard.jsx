import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Button, Card } from 'semantic-ui-react';

const extra = <Button type="submit" className="details">Details</Button>;

const ProductsCard = ({ products }) => {
  if (products) {
    return (
      products.map(product => (
        <Link to={`/products/${product.productid}`} key={product.productid}>
          <Card
          key={product.productid}
          className="products"
          image={product.productimage}
          header={product.productname}
          meta="The most amazing product money can buy"
          description={product.price}
          extra={extra}
        />
        </Link>
      ))
    );
  }
  return <h2 className="noProduct">There are presently no products in the store!</h2>;
};

ProductsCard.propTypes = {
  products: PropTypes.array
};
ProductsCard.defaultProps = {
  products: []
};

export default ProductsCard;
