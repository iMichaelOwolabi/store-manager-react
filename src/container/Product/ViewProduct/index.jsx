import React, { Fragment } from 'react';
import AuthNav from '../../../components/AuthNav';

const ViewProduct = () => (
  <Fragment>
    <AuthNav />
    <div className="container">
      <h1>Product Details</h1>
    </div>
  </Fragment>
);
export default ViewProduct;
