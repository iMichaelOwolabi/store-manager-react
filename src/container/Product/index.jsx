import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import { Button } from 'semantic-ui-react';
import AuthNav from '../../components/AuthNav';
import './Product.scss';

const Product = () => (
  <Fragment>
    <AuthNav />
    <div className="container">
      <Link to="/create-product">
        <Button basic color="violet" content="Add Product" className="newProduct" />
      </Link>
    </div>
  </Fragment>
);
export default Product;
