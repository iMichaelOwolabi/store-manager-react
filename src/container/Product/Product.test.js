import React from 'react';
import { shallow } from 'enzyme';
import Product from '.';

describe('Test the 404 Component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Product />);
    expect(wrapper).toMatchSnapshot();
  });
});
