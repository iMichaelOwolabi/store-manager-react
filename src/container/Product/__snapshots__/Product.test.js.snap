// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the 404 Component should match snapshot 1`] = `<theProduct />`;
